### PR TITLE
Prometheus: Frontend package fixes for using in external vendor DS

### PR DESCRIPTION
**Pull request by bohandley**
**What is this?**

This updates the prometheus library so we can use it in the external vendor data source plugin. Currently the library works nicely for core Grafana but without these changes it can not be used in an external Prometheus data source. Therefore, we need these changes to test in the external vendor data source.

- [x] consistent exports, no nested barrel files and explicitly named exports
- [x] only export what is necessary for the external vendor DS 
- [x] use rollup to bundle correctly for external vendor DS plugin
- [x] update license in package.json to be AGPL
- [x] keep package.json library listed as private until we're done testing and ready to actually publish
- [x] fix stateSlice typing issue
- [x] fix monaco loader issue in external DS
- [x] default to label matchers support with removed version detection (https://github.com/grafana/grafana/issues/81059)

**Outside this PR scope**
- [ ] Remove gcopypaste folder
- [ ] Use the shared/ folder that has been migrated to grafana/experimental

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-prometheus/rollup.config.ts
Comment (jackw) [inline packages/grafana-prometheus/rollup.config.ts:9]: I'm interested in understanding why there is a need to bundle all the icons in this package? Icon usage comes from the `<Icon />` component in `@grafana/ui` which in turn already uses the contents of this cache via Grafana core.

File: packages/grafana-prometheus/package.json
Comment (jackw) [inline packages/grafana-prometheus/package.json:84]: Not sure this is needed. I cannot see it being used in the rollup.config file.

File: packages/grafana-prometheus/package.json
Comment (jackw) [inline packages/grafana-prometheus/package.json:86]: Not sure this is needed. I cannot see it being used in the rollup.config file.

File: packages/grafana-prometheus/src/components/index.ts
Comment (jackw) [inline packages/grafana-prometheus/src/components/index.ts:11]: Probably not something for a follow up PR... for clarities sake, and to prevent accidentally exposing anything you'd like to keep private, I would avoid all nested barrel files entirely and use named exports rather than these wildcards in `src/index.ts`.

File: packages/grafana-prometheus/rollup.config.ts
Comment (bohandley) [reply packages/grafana-prometheus/rollup.config.ts:9]: I can remove the icons, those were left over from the PromQAIL work and needed to be moved. We don't have to have them in the package. I'll get a PR for that up today.

**Review by itsmylife — COMMENTED** [review]
LGTM but I think Jack should be the final approver

File: packages/grafana-prometheus/src/components/index.ts
Comment (bohandley) [reply packages/grafana-prometheus/src/components/index.ts:11]: I can definitely do this!

File: packages/grafana-prometheus/package.json
Comment (bohandley) [reply packages/grafana-prometheus/package.json:86]: Removing!

File: packages/grafana-prometheus/src/components/index.ts
Comment (bohandley) [reply packages/grafana-prometheus/src/components/index.ts]: I have removed the nested barrel files I'm exporting with names instead of wildcards. I am aiming to be exporting just enough to make the custom auth work for our external vendor data sources. That will make sure that we aren't exporting anything private for now.

File: packages/grafana-prometheus/src/components/monaco-query-field/promql.ts
Comment (jackw) [inline packages/grafana-prometheus/src/components/monaco-query-field/promql.ts:14]: Have you confirmed if this will be stripped when bundling? Not sure what implications it has. 🤔

Comment (bohandley) [reply packages/grafana-prometheus/src/components/monaco-query-field/promql.ts:14]: I can confirm it is stripped when bundling. I know having the MIT license allows any further usage to copy the code and change the license though, so when it becomes AGPL in the library, that should be ok.

File: packages/grafana-prometheus/package.json
Comment (jackw) [inline packages/grafana-prometheus/package.json:6]: This should match the other packages version across Grafana.
```suggestion
  "version": "10.4.0-pre",
```

File: packages/grafana-prometheus/package.json
Comment (jackw) [inline packages/grafana-prometheus/package.json:36]: Not sure this icons script is needed.

File: packages/grafana-prometheus/package.json
Comment (jackw) [inline packages/grafana-prometheus/package.json:5]: You will need to remove this `private` property before you merge this if you want it to be published.

**Review by jackw — APPROVED** [review]
From a frontend-ops perspective... LGTM! 🚀

**Pull request merged**
